### PR TITLE
Add `dotnet vstest` deprecation message

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
image

### CODE REVIEW

**Review by nohwnd — APPROVED** [review]

**Comment by nohwnd** [comment]
From technical point of view it looks good, if our PM approves, then let's go.

File: test/vstest.console.UnitTests/ExecutorUnitTests.cs [hunk 1]
```diff
@@ -338,6 +338,25 @@ public void ExecutorShouldPrintRunnerArchitecture()
         Assert.IsFalse(mockOutput.Messages.Any(message => message.Message!.Contains("vstest.console.exe is running in emulated mode")));
     }
 
+    [TestMethod]
+    [DataRow("--ShowDeprecateDotnetVSTestMessage")]
```
Comment (Evangelink) [inline test/vstest.console.UnitTests/ExecutorUnitTests.cs:342]: I think you should use `DotNet`.

```suggestion
    [DataRow("--ShowDeprecateDotNetVSTestMessage")]
```

File: test/vstest.console.UnitTests/ExecutorUnitTests.cs [hunk 2]
```diff
@@ -338,6 +338,25 @@ public void ExecutorShouldPrintRunnerArchitecture()
...
+    [DataRow("--showdeprecateDotnetvTestMessage")]
```
Comment (Evangelink) [inline test/vstest.console.UnitTests/ExecutorUnitTests.cs:343]: Probably:

```suggestion
    [DataRow("--showDeprecateDotNetVSTestMessage")]
```

File: test/vstest.console.UnitTests/ExecutorUnitTests.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply test/vstest.console.UnitTests/ExecutorUnitTests.cs:342]: Here `Dotnet` is the muxer `dotnet`, interesting...I've always used dotnet as single word

File: test/vstest.console.UnitTests/ExecutorUnitTests.cs (hunk 2 above)
Comment (MarcoRossignoli) [reply test/vstest.console.UnitTests/ExecutorUnitTests.cs:343]: it's done on purpose to test the ordinal ignore casing.

Comment (MarcoRossignoli) [reply test/vstest.console.UnitTests/ExecutorUnitTests.cs:343]: never mind me...I'm old.

File: test/vstest.console.UnitTests/ExecutorUnitTests.cs (hunk 1 above)
Comment (nohwnd) [reply test/vstest.console.UnitTests/ExecutorUnitTests.cs:342]: What makes you think that? The product name is dotnet or dotNET, with the first one being used way more often imo. So Dotnet is correct. DotNET would also be correct, but dotNet is discouraged.

This guideline says to note the capitalization (and not spell with a space in the middle). 

https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/n/net

This brand guide does not mention the spelling we are using `dotnet` explicitly, but uses it throughout the guide: https://github.com/dotnet/brand

Comment (nohwnd) [reply test/vstest.console.UnitTests/ExecutorUnitTests.cs:342]: https://github.com/dotnet/brand/issues/26 added issue for that.

**Pull request merged**